File: JavaScript/programmers/가장큰수.js
function solution(numbers) {
  var answer = "";
  const arr = [];
  for (let i = 0; i < numbers.length; i++) {
    arr.push(numbers[i].toString());
  }

  arr.sort((a, b) => b + a - (a + b));
  for (let num of arr) {
    answer += num;
  }

  if (answer[0] === "0") return "0";

  return answer;
}

File: javaalgo/programmers/정렬/가장큰수.java
package javaalgo.programmers.정렬;

import java.util.Arrays;

public class 가장큰수 {
	 public String solution(int[] numbers) {
		    String[] arr = new String[numbers.length];
	        for(int i=0; i<numbers.length; i++){
	            arr[i] = Integer.toString(numbers[i]);
	        }
	        String answer="";
	        Arrays.sort(arr,(a,b)->{
	            return (a+b).compareTo(b+a);
	        }); 
	        for(int i=arr.length-1; i>=0.; i--){
	            answer+=arr[i];
	        }
	        if(answer.charAt(0) == '0') answer = "0";
	        
	        return answer;
	    }
}
